byterun/caml/alloc.h

```diff
@@ -58,12 +70,15 @@ CAMLextern int caml_convert_flag_list (value, int *);
 /* Convenience functions to deal with unboxable types. */
 static inline value caml_alloc_unboxed (value arg) { return arg; }
 static inline value caml_alloc_boxed (value arg) {
-  value result = caml_alloc_small (1, 0);
-  Field (result, 0) = arg;
-  return result;
+  return caml_alloc_1(0, arg);
 }
 static inline value caml_field_unboxed (value arg) { return arg; }
-static inline value caml_field_boxed (value arg) { return Field (arg, 0); }
+static inline value caml_field_boxed (value arg) {
+  CAMLparam1(arg);
+  CAMLlocal1(v);
+  caml_read_field(arg, 0, &v);
+  CAMLreturn (v);
+}
```

Comment:
You're right, of course: if `caml_read_field` faults into the GC (on multicore) then it's only `arg` that will need protecting in this case. Still, I'd prefer for my version to be the documented advice, since it's an easy rule to conform to (even if it is a little conservative at times).